sklearn/preprocessing/_encoders.py

```diff
@@ -110,7 +110,16 @@ def _transform(self, X, handle_unknown='error'):
                     # continue `The rows are marked `X_mask` and will be
                     # removed later.
                     X_mask[:, i] = valid_mask
-                    Xi = Xi.copy()
+                    # cast Xi into the largest string type
```

Comment:
Can you mention here as well that this is only about numpy string dtypes? (and not strings in an object array?)